launcher/src/main/kotlin/org/sourcegrade/jagr/launcher/executor/ProgressBar.kt

```diff
@@ -51,6 +57,16 @@ class ProgressBar(
     if (progressDecimal < 1.0) {
       sb.append(tipChar)
     }
+
+    if (rainbowProgressBar) {
+      val tmp = StringBuilder(30)
```

Comment:
It should be possible to calculate the exact buffer size needed based on the existing string builder.

Reply:
The buffer size only needed for the tmp StringBuilder or the whole String containing the percentage at the end? The capacity for the previous StringBuilder sb is also too small.